engineio/src/client/client.rs

```diff
@@ -128,7 +129,8 @@ impl ClientBuilder {
 
         let mut url = self.url.clone();
 
-        let handshake: HandshakePacket = Packet::try_from(transport.poll()?)?.try_into()?;
+        let handshake: HandshakePacket =
+            Packet::try_from(transport.poll(DEFAULT_MAX_POLL_TIMEOUT)?)?.try_into()?;
```

Comment:
`DEFAULT_MAX_POLL_TIMEOUT` is only used here, before there was no timeout. Should this be configurable or without a timeout here?

Reply:
Hmmm, hard to say.

I needed a timeout here, because I changed the signature of the method to include a timeout.

- We could make the timeout optional but then we'd have to check that everywhere.
- We could use a very large timeout (so something like infinity, but there is not `Duration::infinity` or similar) to simulate no timeout. That number would seem arbitrary, though.
- We could make it configurable but I'm unsure whether that would actually be of use for someone.

Personally, I think the current timeout of 45 seconds (the default maximum ping timeout) is reasonable for a handshake to complete.

Reply:
Ok, sounds reasonable.